### PR TITLE
[Scala3] Fix `For`/`While` expr that might have indented body without `do` keyword

### PR DESCRIPTION
**Pull request by dos65**

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala [hunk 1]
```diff
@@ -740,6 +740,26 @@ class ControlSyntaxSuite extends BaseDottySuite {
     )
   }
 
+  test("new-for-indented-without-do") {
+    val code = """|for ( a <- x )
```
Comment (tgodzik) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala:744]: Could you also add a test for:
```
  val code = """|for { a <- x }
                  |  fx
                  |  fy
                  |""".stripMargin
```
?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks a lot for the fix! I would just add the test for the last remaining case

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala (hunk 1 above)
Comment (dos65) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala:744]: Added

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**